### PR TITLE
added page to manage user access

### LINKED ISSUE #136: Admin Manage Other User's Access Types

### PR DESCRIPTION
**Pull request by Stuart6557**
# Info

Closes #136

[description]

Added a page that allows admins to update user access types.

## Changes

New page and API

# Type of Change

- [ ] Bug Fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as
      expected)
- [ ] Logistics Change (A change to a README, description, or dev workflow setup like
      linting/formatting)
- [ ] Continuous Integration Change (Related to deployment steps or continuous integration
      workflows)
- [ ] Other: (Fill In) 

# Testing

I have tested that my changes fully resolve the linked issue ...

- [x] locally on Desktop.
- [ ] on the live deployment preview on Desktop.
- [ ] on the live deployment preview on Mobile.
- [ ] I have added new Cypress tests that are passing.

# Checklist

- [x] I have performed a self-review of my own code.
- [ ] I have followed the style guidelines of this project.
- [ ] I have documented any new functions in `/src/lib/*` and commented hard to understand areas
      anywhere else.
- [x] My changes produce no new warnings.

# Screenshots

image

image

### CODE REVIEW

File: .gitignore
Comment (farisashai) [inline .gitignore:1]: let's revert this file since it's unrelated to the PR - this project isn't supposed to be using a .env file anyways. the development and production env files are already included here

File: .vscode/settings.json
Comment (farisashai) [inline .vscode/settings.json:1]: same thing here with reverting the file, make sure you double check your diff in the PR before submitting reviewers because sometimes your editor will automatically make changes on your behalf that get included

File: src/lib/api/index.ts
Comment (farisashai) [inline src/lib/api/index.ts]: let's rename this to AdminAPI to encompass a broader set of admin specific portal actions - right now the name is too specific to the specific route you wrote here

File: src/lib/config.ts
Comment (farisashai) [inline src/lib/config.ts:29]: what is this route for? are we planning to have separate pages to view user access and mange it? I think ideally this should all be on a single page in a table UI eventually or something similar to view all users so since this hasn't been implemented yet, i think it makes more sense to include any view access functionality on the same manageUserAccess page

File: src/pages/admin/manage-user-access/index.tsx
Comment (farisashai) [inline src/pages/admin/manage-user-access/index.tsx]: we don't want to hardcode these options here as magic variables - we have an enum defined in the constants folder I think for `UserAccessType` that you can map through to ensure the options are consistent across the portal

File: src/pages/admin/manage-user-access/index.tsx
Comment (farisashai) [inline src/pages/admin/manage-user-access/index.tsx]:
```suggestion
        name="User Access"
```

File: src/pages/admin/manage-user-access/index.tsx
Comment (farisashai) [inline src/pages/admin/manage-user-access/index.tsx]: we can delete this placeholder since there's a default selected option on initialization

File: src/pages/admin/manage-user-access/index.tsx
Comment (farisashai) [inline src/pages/admin/manage-user-access/index.tsx]: this try catch block should be in a manager function (see src/lib/managers/AuthManager file for an example) to separate client UI logic from business logic interacting with the API

File: src/pages/admin/manage-user-access/index.tsx
Comment (farisashai) [inline src/pages/admin/manage-user-access/index.tsx]: if we're merging this directly into main can you leave a comment that this function needs to be moved to the util class after sean's PR is merged

**Review by farisashai — CHANGES_REQUESTED** [review]
excellent first PR - feature is working with good code quality. just a couple comments

File: src/lib/config.ts
Comment (Stuart6557) [reply src/lib/config.ts:29]: this is an api endpoint, I used it in AdminAPI.ts

File: src/pages/admin/manage-user-access/index.tsx
Comment (Stuart6557) [reply src/pages/admin/manage-user-access/index.tsx]: placeholder is required for SignInFormItem regardless of whether element is "input" or "select", should I change this in SignInFormItem?

**Review by farisashai — COMMENTED** [review]
backend is making a change where admin access levels can't be modified that will need to be reflected on this page too. You'll need to remove admin from the dropdown here when it's merged but we should hold off until then

File: src/lib/config.ts
Comment (alexzhang1618) [inline src/lib/config.ts]: can we just make this `/admin/access` to be consistent

File: src/pages/admin/manage-user-access/index.tsx
Comment (alexzhang1618) [inline src/pages/admin/manage-user-access/index.tsx]: let's merge in main and use the `reportError` in utils

**Review by alexzhang1618 — APPROVED** [review]
changes generally LGTM, let's get this up to date with main and merged in

**Review by farisashai — APPROVED** [review]
i was waiting for https://github.com/acmucsd/membership-portal/pull/407 to merge first so we could implement the feature fully first go but we can do that in a separate PR if there's a hurry to make this available

**Pull request merged**
